api/src/main/java/org/geysermc/geyser/api/item/custom/CustomItemData.java

```diff
@@ -114,6 +115,92 @@ public interface CustomItemData {
      */
     @NonNull Set<String> tags();
 
+    /**
+     * Gets the stack size of the item.
+     *
+     * Returns 0 if not set. When not set (or 0), takes the Java item stack count when based of a vanilla item, or uses 64 when porting a modded item.
+     *
+     * @return the stack size of the item
+     */
+    @NonNegative
+    int stackSize();
+
+    /**
+     * Gets the max damage of the item.
+     *
+     * Returns -1 if not set. When not set (or below 0), takes the Java item max damage when based of a vanilla item, or uses 0 when porting a modded item.
+     *
+     * @return the max damage of the item
+     */
+    int maxDamage();
+
+    /**
+     * Gets the attack damage of the item.
+     * This is purely visual, and only applied to tools
+     *
+     * Returns 0 if not set. When 0, takes the Java item attack damage when based of a vanilla item, or uses 0 when porting a modded item.
+     *
+     * @return the attack damage of the item
+     */
+    int attackDamage();
+
+    /**
+     * Gets the tool type of the item.
+     *
+     * @return the tool type of the item
+     */
+    @Nullable String toolType();
+
+    /**
+     * Gets the tool tier of the item.
+     *
+     * @return the tool tier of the item
+     */
+    @Nullable String toolTier();
+
+    /**
+     * Gets the armor type of the item.
+     *
+     * @return the armor type of the item
+     */
+    @Nullable String armorType();
+
+    /**
+     * Gets the armor protection value of the item.
+     *
+     * @return the armor protection value of the item
+     */
+    int protectionValue();
+
+    /**
+     * Gets if the item is a hat. This is used to determine if the item should be rendered on the player's head, and
+     * normally allow the player to equip it. This is not meant for armor.
+     *
+     * @return if the item is a hat
+     */
+    boolean isHat();
```

Comment:
I'm not convinced that this (and all other moved api that isn't controlled by a component) should be moved here. lMO it makes more sense to inherit this from the base Java item, as is done in vanilla

Reply:
Hm, I was a bit conflicted on some of the moves.
The only properties that don't relate to a Java component at all are, I think, `toolType`, `toolTier`, `armorType` and `isHat`.

As for `toolType` and `toolTier`, there is a `minecraft:tool` component on Java, but I'm not sure what the functions of the `toolType` and `toolTier` properties are, and if/how they relate to the Java `minecraft:tool` component.

Same goes for the `armorType` property, though there isn't an armor component on Java - rather, armor attribute modifiers are in the `minecraft:attribute_modifiers` component.

As for `isHat`, I moved this because I think it can be useful for custom helmets making use of vanilla items with custom models. On the Java helmet slot, all vanilla items are always rendered, even with custom models, which allows for custom helmet models, while on Bedrock this requires the `isHat` property.
Additionally, mods like Polymer allow adding custom helmet items (appearing as vanilla items to the client) that can be equipped in the helmet slot on a vanilla Java client, so I thought it would be useful to copy this to allow for the same behaviour on a Bedrock client.

Reply:
Which vanilla item does polymer use for hats on Java? If it's skulls/carved pumpkins, that could probably be set by us too

Reply:
Polymer can use any for hats - since it's a library mods use to make items server side only, the mod using it can choose which item appears on the client side.

Reply:
Okay, let's leave armor type/isHat as is then